pkg/archive/archive_test.go

```diff
@@ -724,6 +726,65 @@ func TestTarUntar(t *testing.T) {
 	}
 }
 
+func TestTarAppenderChownOption(t *testing.T) {
+	origin, err := ioutil.TempDir("", "docker-test-tar-chown-opt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(origin)
+	if err := ioutil.WriteFile(filepath.Join(origin, "1"), []byte("hello world"), 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	pipeReader, pipeWriter := io.Pipe()
+	errCh := make(chan error)
+	go func() {
+		ta := newTarAppender(
+			&idtools.IDMappings{},
+			pipeWriter,
+			&idtools.IDPair{UID: 1337, GID: 42},
+		)
+
+		defer func() {
+			if err := ta.TarWriter.Close(); err != nil {
+				errCh <- err
+			}
+			if err := pipeWriter.Close(); err != nil {
+				errCh <- err
+			}
+		}()
+		defer pools.BufioWriter32KPool.Put(ta.Buffer)
+
+		if err := ta.addTarFile(filepath.Join(origin, "1"), "1"); err != nil {
```

Comment:
@tonistiigi I can't figure out why in `addTarFile` the way I've written this test code the write of headers in https://github.com/moby/moby/blob/f291c959ef6294757db92be5e2add774ceda60cc/pkg/archive/archive.go#L495 gets frozen so it leads to a fatal timeout ATM. Excepting that, is it the kind of expected test? Will add some test cases.